server/src/main/java/org/apache/calcite/avatica/server/ServerCustomizer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.avatica.server;
+
+import org.eclipse.jetty.server.Server;
+
+/**
+ * Callback for customizing a Jetty {@link Server}.
+ */
+public interface ServerCustomizer {
+  /**
+   * Customize the server during initialization.
+   * @param server The server to customize
+   */
+  void customize(Server server);
```

Comment:
I think this should be changed -- perhaps parameterize this class with the concrete `Server` type. Avatica's API should not strictly depend on an implementation details (in this case, that Jetty is being used). We don't want users to know that Jetty is the webserver used behind the scenes.

e.g.

```java
public interface ServerCustomizer<T> {
  void customize(T server);
}
```

Making this change would also require some changes to go into the HttpServer.Builder class to make sure that users know exactly what they're getting. I think deprecating `HttpServer` and making it `JettyHttpServer` instead is the right avenue (or something along those lines).

Does that make sense?